### PR TITLE
Dịch "chapter_preliminaries/autograd_vn" - Phần 05

### LINKED ISSUE #471: Dịch "chapter_preliminaries/autograd_vn" - Phần 05
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_preliminaries/autograd_vn.md#L269-L346

### PR DESCRIPTION
**Pull request by thduynguyen**
## Pull Request checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #471 

* [x] Bản dịch này có bám sát [Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md) không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm
    * Tiêu đề (số lượng dấu `#` đầu dòng)
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`)
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt)
    * Không dịch các phần code (nằm giữa hai dấu `````)
    * Copy các công thức toán từ bản gốc (các đoạn có `$`)
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`)
    * Không tự thêm bớt các dòng trắng
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm [hướng dẫn hoặc yêu cầu dịch hình](../img/README.md).

* [x] Điền tên của bạn và vào mục "Những người thực hiện" ở cuối file.

* [x] Pull Request này đã sẵn sàng cho việc Review chưa?

* [ ] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này.

## Với reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_preliminaries/autograd_vn.md [hunk 1]
```diff
@@ -291,7 +291,9 @@ and only take into account the role
 that `x` played after `y` was calculated.
 -->
 
-*dịch đoạn phía trên*
+Đôi khi chúng ta muốn chuyển một số phép tính ra khỏi đồ thị tính toán.
+Ví dụ như trong trường hợp ta đã tính `y` như là một hàm của `x`, rồi sau đó tính `z` như là một hàm của cả `y` và `x`.
+Bây giờ, giả dụ như ta muốn tính gradient của `z` theo `x`, nhưng vì lý do nào đó ta lại muốn xem `y` như là một hằng số, và chỉ xét đến vai trò của `x` như là biến số của `z` sau khi giá trị của `y` đã được tính.
```
Comment (tiepvupsu) [inline chapter_preliminaries/autograd_vn.md]:
```suggestion
Bây giờ, giả sử ta muốn tính gradient của `z` theo `x`, nhưng vì lý do nào đó ta lại muốn xem `y` như là một hằng số và chỉ xét đến vai trò của `x` như là biến số của `z` sau khi giá trị của `y` đã được tính.
```

File: chapter_preliminaries/autograd_vn.md [hunk 2]
```diff
@@ -291,7 +291,9 @@ and only take into account the role
 that `x` played after `y` was calculated.
 -->
 
-*dịch đoạn phía trên*
+Đôi khi chúng ta muốn chuyển một số phép tính ra khỏi đồ thị tính toán.
+Ví dụ như trong trường hợp ta đã tính `y` như là một hàm của `x`, rồi sau đó tính `z` như là một hàm của cả `y` và `x`.
```
Comment (tiepvupsu) [inline chapter_preliminaries/autograd_vn.md]:
```suggestion
Ví dụ, giả sử `y` đã được tính như một hàm của `x`, rồi sau đó `z` được tính như một hàm của cả `y` và `x`.
```

File: chapter_preliminaries/autograd_vn.md [hunk 3]
```diff
@@ -306,7 +308,10 @@ the partial derivative of `z = u * x` with respect to `x` while treating `u` as
 instead of the partial derivative of `z = x * x * x` with respect to `x`.
 -->
 
-*dịch đoạn phía trên*
+Trong trường hợp này, ta có thể gọi `u = y.detach()` để trả về một biến `u` mới có cùng giá trị như `y` nhưng loại bỏ ra các thông tin về cách mà `y` đã được tính trong đồ thị tính toán.
```
Comment (minhduc0711) [inline chapter_preliminaries/autograd_vn.md]: Câu này bản tiếng anh nghe cứ sai sai
```suggestion
Trong trường hợp này, ta có thể gọi `u = y.detach()` để trả về một biến `u` mới có cùng giá trị như `y` nhưng không còn chứa các thông tin về cách mà `y` được tính trong đồ thị tính toán.
```

File: chapter_preliminaries/autograd_vn.md [hunk 4]
```diff
@@ -306,7 +308,10 @@ the partial derivative of `z = u * x` with respect to `x` while treating `u` as
...
+Nói cách khác, gradient sẽ không thể chảy ngược qua `u` để đến `x` được.
```
Comment (minhduc0711) [inline chapter_preliminaries/autograd_vn.md]:
```suggestion
Nói cách khác, gradient sẽ không thể chảy ngược qua `u` về `x` được.
```

File: chapter_preliminaries/autograd_vn.md [hunk 5]
```diff
@@ -306,7 +308,10 @@ the partial derivative of `z = u * x` with respect to `x` while treating `u` as
...
+Bằng cách này, ta đã tính `u` như là một hàm của `x` ở ngoài phạm vi của `autograd.record`, dẫn đến việc biến `u` sẽ được xem như là một hằng số mỗi khi ta gọi `backward`.
```
Comment (minhduc0711) [inline chapter_preliminaries/autograd_vn.md]:
```suggestion
Bằng cách này, ta đã tính `u` như một hàm của `x` ở ngoài phạm vi của `autograd.record`, dẫn đến việc biến `u` sẽ được xem như là một hằng số mỗi khi ta gọi `backward`.
```

File: chapter_preliminaries/autograd_vn.md [hunk 6]
```diff
@@ -306,7 +308,10 @@ the partial derivative of `z = u * x` with respect to `x` while treating `u` as
...
+Chính vì vậy, hàm `backward` sau đây sẽ tính đạo hàm riêng của `z = u * x` theo `x` trong khi xem `u` như là một hằng số, thay vì tính đạo hàm riêng của `z = x * x * x` theo `x`.
```
Comment (minhduc0711) [inline chapter_preliminaries/autograd_vn.md]:
```suggestion
Chính vì vậy, hàm `backward` sau đây sẽ tính đạo hàm riêng của `z = u * x` theo `x` khi xem `u` như là một hằng số, thay vì đạo hàm riêng của `z = x * x * x` theo `x`.
```

File: chapter_preliminaries/autograd_vn.md [hunk 7]
```diff
@@ -525,7 +531,7 @@ với dấu `@` ở đầu. Ví dụ: @aivivn.
 *
 
 <!-- Phần 5 -->
-*
+* Tạ H. Duy Nguyên
```
Comment (thduynguyen) [inline chapter_preliminaries/autograd_vn.md]:
```suggestion
* Tạ H. Duy Nguyên
* Vũ Hữu Tiệp
* Phạm Minh Đức
```
